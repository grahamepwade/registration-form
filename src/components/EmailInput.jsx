import React, { useState } from 'react'

import FormControl from '@mui/material/FormControl'
import FormLabel from '@mui/material/FormLabel'
import TextField from '@mui/material/TextField'

import makeStyles from '@mui/styles/makeStyles'

import validator from 'validator'

const useStyles = makeStyles((theme) => ({
    emailInputStyle: {
        width: '100%',
        color: theme.palette.secondary.main,
    },
    formLabelStyle: {
        paddingBottom: '5px',
        color: theme.palette.secondary.main
    },
    textFieldInputStyle: {
        '& p': {
            color: 'red',
            marginLeft: 0
        }
    }
}))


const EmailInput = ({ handleEmailAddress }) => {
    const [email, setEmail] = useState({ errorText: '', value: '' })

    const classes = useStyles()

    const handleEmailChange = (input) => {
        let emailObj = {...email}
        emailObj.value = input

        setEmail(emailObj)
    }

    const handleEmailBlur = () => {
        let emailObj = {...email}

        if (emailObj.value === '') {
            emailObj.errorText = 'Cannot be empty'
        }
        else if (validator.isEmail(emailObj.value)) {
            emailObj.errorText = ''
        }
        else {
            emailObj.errorText = 'Invalid input'
        }
        
        setEmail(emailObj)
        handleEmailAddress(emailObj)
    }


    return (
        <FormControl className={classes.emailInputStyle}>
            <FormLabel className={classes.formLabelStyle} component='legend'>Email address</FormLabel>
            <TextField 
                className={classes.textFieldInputStyle}
                inputProps={{style: {fontSize: 'small'}}}
                fullWidth={true}
                size='small' 
                label='' 
                id='email' 
                type='text' 
                placeholder='Email address'
                value={email.value}
                onChange={e => handleEmailChange(e.target.value)}
                onBlur={handleEmailBlur}
                helperText={email.errorText}
            />
        </FormControl>
    )
}

export default EmailInput