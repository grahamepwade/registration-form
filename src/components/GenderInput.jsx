import React, { useState } from 'react'

import Button from '@mui/material/Button'
import FormControl from '@mui/material/FormControl'
import FormLabel from '@mui/material/FormLabel'

import makeStyles from '@mui/styles/makeStyles'
import Container from '@mui/material/Container'

const useStyles = makeStyles((theme) => ({
    genderFormStyle: {
        width: '100%',
    },
    formLabelStyle: {
        paddingBottom: '5px',
        color: theme.palette.secondary.main,
    },
    buttonContainerStyle: {
        paddingLeft: '0'
    },
    buttonStyle: {
        fontSize: '13px',
        boxShadow: 'none',
        textTransform: 'none',
        minWidth: '5rem',
        height: '34px',
        padding: '5px 15px',
        marginRight: '20px',
        color: theme.palette.secondary.main + ' !important',
        borderColor: 'rgba(0, 0, 0, 0.23)',
        '&:hover': {
            borderColor: theme.palette.primary.main + ' !important',
        }
    },
}))


const GenderInput = ({ handleGender }) => {
    const [gender, setGender] = useState({ errorText: '', value: '' })

    const classes = useStyles()

    const handleChange = (selection) => {
        let genderObj = {...gender}
        genderObj.value = selection

        handleGender(genderObj)
        setGender(genderObj)
    }

    return (
        <FormControl className={classes.genderFormStyle}>
            <FormLabel className={classes.formLabelStyle} component='legend'>Gender</FormLabel>
            <Container className={classes.buttonContainerStyle}>
                <Button
                    className={classes.buttonStyle}
                    variant= {gender.value === 'male' ? 'contained' : 'outlined'}
                    style={{backgroundColor: gender.value === 'male' ? '#00b79b' : ''}}
                    size='small'
                    value='male'
                    onClick={() => handleChange('male')}
                >
                    Male
                </Button>
                <Button
                    className={classes.buttonStyle}
                    value='female'
                    variant= {gender.value === 'female' ? 'contained' : 'outlined'}
                    style={{backgroundColor: gender.value === 'female' ? '#00b79b' : ''}}
                    size='small'
                    onClick={() => handleChange('female')}
                >
                    Female
                </Button>
                <Button
                    className={classes.buttonStyle}
                    value='other'
                    variant= {gender.value === 'other' ? 'contained' : 'outlined'}
                    style={{backgroundColor: gender.value === 'other' ? '#00b79b' : ''}}

                    size='small'
                    onClick={() => handleChange('other')}
                >
                    Other
                </Button>
            </Container>
        </FormControl>
    )
}

export default GenderInput