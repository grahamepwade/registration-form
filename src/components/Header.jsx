import React from 'react'

import AppBar from '@mui/material/AppBar'
import Button from '@mui/material/Button'
import Toolbar from '@mui/material/Toolbar'
import Box from '@mui/system/Box'

import makeStyles from '@mui/styles/makeStyles'

import logo from 'fe_test/logo.svg'

const useStyles = makeStyles((theme) => ({
    header: {
        backgroundColor: '#fff !important',
        boxShadow: 'none !important',
    },
    logoStyle: {
        width: '-webkit-fill-available',
        height: '50px',
        backgroundImage: `url(${logo})`,
        backgroundRepeat: 'no-repeat',
        color: theme.palette.primary.main
    },
    headerButtonStyle: {
        textTransform: 'none',
        minWidth: 'fit-content',
        color: theme.palette.primary.main +  ' !important',
        borderColor: theme.palette.primary.main  +  ' !important',
        '&:hover': {
            borderColor: theme.palette.primary.main,
        }
    }
}))

const Header = () => {

    const classes = useStyles()

    return (
        <AppBar className={classes.header} position='static'>
            <Toolbar variant='regular'>
                <Box className={classes.logoStyle} />
                <Button className={classes.headerButtonStyle} variant='outlined' size='small'>
                    How it works
                </Button>
            </Toolbar>
        </AppBar>
    )
}

export default Header