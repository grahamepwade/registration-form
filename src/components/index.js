export { default as Header } from './Header'
export { default as FirstNameInput } from './FirstNameInput'
export { default as LastNameInput } from './LastNameInput'
export { default as EmailInput } from './EmailInput'
export { default as StreetInput } from './StreetInput'
export { default as CityInput } from './CityInput'
export { default as TerritoryInput } from './TerritoryInput'
export { default as ZIPInput } from './ZIPInput'
export { default as PhoneInput } from './PhoneInput'
export { default as DOBInput } from './DOBInput'
export { default as GenderInput } from './GenderInput'
export { default as SubmitButton } from './SubmitButton'