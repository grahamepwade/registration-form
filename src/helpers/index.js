export { default as sendFormData } from './fetch'

export { default as formatPhoneNumber } from './formatPhoneNumber'
export { default as statesProvincesList} from './statesProvincesList'

export { default as validCity } from './validCity'
export { default as validDOB } from './validDOB'
export { default as validName } from './validName'
export { default as validStreet } from './validStreet'
export { default as validZIP } from './validZIP'
