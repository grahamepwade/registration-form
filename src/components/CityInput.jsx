import React, { useState } from 'react'

import FormControl from '@mui/material/FormControl'
import FormLabel from '@mui/material/FormLabel'
import TextField from '@mui/material/TextField'

import makeStyles from '@mui/styles/makeStyles'

import { validCity } from 'helpers'
 
const useStyles = makeStyles((theme) => ({
    cityInputStyle: {
        width: '100%',
        color: theme.palette.secondary.main,
    },
    formLabelStyle: {
        paddingBottom: '5px',
        color: theme.palette.secondary.main
    },
    textFieldInputStyle: {
        '& p': {
            color: 'red',
            marginLeft: 0
        }
    }
}))


const CityInput = ({ handleCity }) => {
    const [city, setCity] = useState({ errorText: '', value: '' })

    const classes = useStyles()

    const handleCityChange = (input) => {
        let cityObj = {...city}
        cityObj.errorText = input.match(validCity) === null && input !== '' ? 'Invalid input' : ''
        cityObj.value = input

        setCity(cityObj)
    }

    const handleOnBlur = () => {
        let cityObj = {...city}
        cityObj.errorText = cityObj.value === '' ? cityObj.errorText = 'Cannot be empty' : cityObj.errorText        
        
        setCity(cityObj)
        handleCity(cityObj)
    }

    return (
        <FormControl className={classes.cityInputStyle}>
            <FormLabel className={classes.formLabelStyle} component='legend'>City</FormLabel>
            <TextField 
                fullWidth={true}
                className={classes.textFieldInputStyle}
                inputProps={{style: {fontSize: 'small'}}}
                size='small' 
                label='' 
                id='city' 
                type='text' 
                placeholder='City'
                value={city.value}
                onChange={e => handleCityChange(e.target.value)}
                onBlur={handleOnBlur}
                helperText={city.errorText}
            />
        </FormControl>
    )
}

export default CityInput