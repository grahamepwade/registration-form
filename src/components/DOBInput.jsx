import React, { useState } from 'react'

import FormControl from '@mui/material/FormControl'
import FormHelperText from '@mui/material/FormHelperText'
import FormLabel from '@mui/material/FormLabel'
import TextField from '@mui/material/TextField'

import AdapterDateFns from '@mui/lab/AdapterDateFns';
import DesktopDatePicker from '@mui/lab/DesktopDatePicker'
import LocalizationProvider from '@mui/lab/LocalizationProvider';

import makeStyles from '@mui/styles/makeStyles'

import { getDate, getMonth, getYear } from 'date-fns'

import { validDOB } from 'helpers'

const useStyles = makeStyles((theme) => ({
    dateInputStyle: {
        width: '100%',
        color: theme.palette.secondary.main,
    },
    formLabelStyle: {
        paddingBottom: '5px',
        color: theme.palette.secondary.main
    },
    datePickerInputStyle: {
        height: '36px',
        fontSize: 'small',
    }
}))


const DOBInput = ({ handleDOB }) => {
    var today = new Date()
    var cutOffDate = new Date(getYear(today) - 18, getMonth(today), getDate(today))

    const [date, setDate] = useState({ errorText: '', value: cutOffDate })

    const classes = useStyles()

    const handleDateChange = (input) => {
        let dateObj = {...date}
        dateObj.value = input

        dateObj.errorText = validDOB(dateObj.value) ? '' : 'Must be 18 or older'

        setDate(dateObj)
        handleDOB(dateObj)
    }

    return (
        <FormControl className={classes.dateInputStyle}>
            <LocalizationProvider dateAdapter={AdapterDateFns}>
                <FormLabel className={classes.formLabelStyle} component='legend'>Date of birth</FormLabel>
                <DesktopDatePicker
                    value={date.value}
                    InputProps={{className: classes.datePickerInputStyle}}
                    onChange={handleDateChange}
                    allowSameDateSelection={true}
                    renderInput={(params) => <TextField {...params} />}
                />
                <FormHelperText 
                    style={{
                        display: date.errorText === '' ? 'none' : 'block',
                        color: 'red',
                        marginLeft: 0
                    }}
                >
                    Must be 18yrs or older
                </FormHelperText>
            </LocalizationProvider>
        </FormControl>
    )
}

export default DOBInput