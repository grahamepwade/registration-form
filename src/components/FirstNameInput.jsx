import React, { useState } from 'react'

import FormControl from '@mui/material/FormControl'
import FormLabel from '@mui/material/FormLabel'
import TextField from '@mui/material/TextField'

import makeStyles from '@mui/styles/makeStyles'

import { validName } from 'helpers'

const useStyles = makeStyles((theme) => ({
    formLabelStyle: {
        paddingBottom: '5px',
        color: theme.palette.secondary.main
    },
    nameInputStyle: {
        width: '100%',
    },
    textFieldInputStyle: {
        '& p': {
            color: 'red',
            marginLeft: 0,
        }
    }
}))


const FirstNameInput = ({ handleFirstName }) => {
    const [firstName, setFirstName] = useState({ errorText: '', value: '' })

    const classes = useStyles()

    const handleFirstNameChange = (input) => {
        let firstNameObj = {...firstName}
        firstNameObj.errorText = input.match(validName) === null && input !== '' ? 'Invalid input' : ''
        firstNameObj.value = input

        setFirstName(firstNameObj)
    }

    const handleOnBlur = () => {
        let firstNameObj = {...firstName}
        firstNameObj.errorText = firstNameObj.value === '' ? firstNameObj.errorText = 'Cannot be empty' : firstNameObj.errorText
        
        setFirstName(firstNameObj)
        handleFirstName(firstNameObj)
    }

    return (
        <FormControl className={classes.nameInputStyle}>
            <FormLabel className={classes.formLabelStyle} component='legend'>First name</FormLabel>
            <TextField 
                className={classes.textFieldInputStyle}
                inputProps={{style: {fontSize: 'small'}}}
                size='small' 
                label='' 
                id='first-name' 
                type='text' 
                placeholder='First name'
                value={firstName.value}
                onChange={e => handleFirstNameChange(e.target.value)}
                onBlur={handleOnBlur}
                helperText={firstName.errorText}
            />
        </FormControl>
    )
}

export default FirstNameInput