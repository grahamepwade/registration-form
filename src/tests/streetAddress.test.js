import { shallow } from 'enzyme';

import { ThemeProvider } from '@mui/styles'
import { createTheme } from '@mui/material/styles'

import { validStreet } from 'helpers'
import { StreetInput } from 'components'

const theme = createTheme({
    palette: {
        primary: {
            main: '#00b79b',
        },
        secondary: {
            main: '#222b3b',
        }
    },
})

const wrapper = shallow(<ThemeProvider theme={theme}> <StreetInput /> </ThemeProvider>);

describe('<StreetInput />', () => {
  it('should work', () => {
    expect(wrapper).toMatchSnapshot()
  });
});


it('validates the street input correctly with characters', () => {
    const input = 'Main'
    expect(input).toMatch(validStreet)
});

it('validates the street input correctly with whitespace', () => {
  const input = 'Main Street'
  expect(input).toMatch(validStreet)
});

it('validates the street input correctly with numbers', () => {
  const input = 'Main21'
  expect(input).not.toMatch(validStreet)
});

it('validates the street input correctly with hyphen', () => {
  const input = 'Main-Street'
  expect(input).toMatch(validStreet)
});

it('validates the street input correctly with period', () => {
  const input = 'Main.Street'
  expect(input).toMatch(validStreet)
});

it('validates the street input correctly with apostrohpe', () => {
  const input = "Main'Street"
  expect(input).not.toMatch(validStreet)
});

it('validates the street input correctly with combination', () => {
  const input = "Main-Street."
  expect(input).toMatch(validStreet)
});

it('validates the street input correctly with combination and whitespace', () => {
  const input = "Main. Street-"
  expect(input).toMatch(validStreet)
});