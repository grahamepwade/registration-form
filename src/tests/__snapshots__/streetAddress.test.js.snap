// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StreetInput /> should work 1`] = `ShallowWrapper {}`;
